benchmark/utils/loggers.hpp

```diff
@@ -186,6 +189,25 @@ struct StorageLogger : gko::log::Logger {
         add_or_set_member(output, "storage", total, allocator);
     }
 
+#if GINKGO_BUILD_MPI
+    void write_data(gko::experimental::mpi::communicator comm,
+                    rapidjson::Value& output,
+                    rapidjson::MemoryPoolAllocator<>& allocator)
+    {
+        const std::lock_guard<std::mutex> lock(mutex);
+        gko::size_type total{};
+        for (const auto& e : storage) {
+            total += e.second;
+        }
+        comm.reduce(gko::ReferenceExecutor::create(),
+                    comm.rank() == 0
+                        ? static_cast<gko::size_type*>(MPI_IN_PLACE)
+                        : &total,
+                    &total, 1, MPI_SUM, 0);
+        add_or_set_member(output, "storage", total, allocator);
```

Comment:
TODO for the future: Figure out if we want to do node-local benchmark results as well. Might be useful for identifying load-imbalance.

Reply:
That's a good point. For the storage, that should be easy enough. But for timings, that would be quite difficult, since we have the synchronization from the global reductions. That is also why I'm just reporting the timings from rank 0 atm, hoping that there is no significant derivation.

Reply:
As long as we collect a runtime breakdown into individual operations, we might still be able to get something from it?

Reply:
Right, in that case it would be helpful. But I would prefer doing the statistics on the different local runtimes in a different PR.

Reply:
definitely!